src/functional/iterator/embedded.py

```diff
@@ -720,18 +719,18 @@ def deref(self) -> Any:
                 raise IndexError("Iterator position doesn't point to valid location for its field.")
         slice_column = dict[Tag, FieldIndex]()
         if self.column_axis is not None:
-            slice_column[self.column_axis] = slice(shifted_pos[self.column_axis], None)
+            assert _column_range is not None
+            k_pos = shifted_pos[self.column_axis]
+            assert isinstance(k_pos, int)
+            slice_column[self.column_axis] = slice(k_pos, k_pos + len(_column_range))
             shifted_pos.pop(self.column_axis)
```

Comment:
```suggestion
            k_pos = shifted_pos.pop(self.column_axis)
            assert isinstance(k_pos, int)
            slice_column[self.column_axis] = slice(k_pos, k_pos + len(_column_range))
```
Feel free to ignore.